### PR TITLE
Setup for Flexible device selection (gpu vs cpu)

### PR DESCRIPTION
**Pull request by khuddzu**
Satisfies Issue #106 
Making changes to workflow so that user can select gpu or cpu for deep learning pytorch workflows. GPU is set to default, but simply adding cpu to training params in yaml file will allow for cpu usage rather than gpu. 

Done:
- Add accelerator and devices to predict workflow in AnvilDeeplearningWorkflow class. This specifies that we will use the same accelerator and device as we used for the training step. 
- Update MTENNSchNetModel predict function to take accelerator and devices as input variables. 

ToDo:
- Update/check chemprop predict function to align with new workflow capabilities
- Add basic test yaml file for cpu usage

### CODE REVIEW

File: openadmet/models/anvil/workflow.py [hunk 1]
```diff
@@ -436,7 +436,9 @@ def run(
         logger.info("Model saved")
 
         logger.info("Predicting")
-        y_pred = self.model.predict(test_dataloader)
+        accelerator = getattr(self.trainer, "accelerator", "gpu")
```
Comment (hmacdope) [inline openadmet/models/anvil/workflow.py:439]: Can you not just use `self.trainer.accelerator.gpu` and `self.trainer.devices`, why use `getattr`?

**Review by hmacdope — CHANGES_REQUESTED** [review]

File: openadmet/models/anvil/workflow.py (hunk 1 above)
Comment (khuddzu) [reply openadmet/models/anvil/workflow.py:439]: self.trainer.accelerator/self.trainer.devices works. (.gpu shouldnt be there). i will change to this as it is cleaner. 
my thought process on getattr was establishing a default again, especially if someone is only predicting, not training, but that is not necessary and is actually a flawed design now that i think about it.

**Comment by khuddzu** [comment]
Ready to merge. 
@hmacdope  maybe a quick last review? 

I did not add a test for this, as none of the current yaml files in test_data use the LightningModule. I'll add to issues and we can add something that goes through the deep learning tests later.

**Comment by hmacdope** [comment]
Just restarted CI, lets see if it finishes

**Comment by hmacdope** [comment]
Im just gonna merge and we can bugfix later.
